include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -488,12 +488,53 @@ inline RunOptions& RunOptions::UnsetTerminate() {
 namespace detail {
 
 template <typename T>
-inline Ort::SessionOptions SessionOptionsImpl<T>::Clone() const {
+std::string ConstSessionOptionsImpl<T>::MakeCustomOpConfigEntryKey(const char* custom_op_name, const char* config) {
+  std::string config_key = "custom_op.";
+
+  config_key += custom_op_name;
+  config_key += ".";
+  config_key += config;
+
+  return config_key;
+}
+
+template <typename T>
+inline Ort::SessionOptions ConstSessionOptionsImpl<T>::Clone() const {
   OrtSessionOptions* out;
   ThrowOnError(GetApi().CloneSessionOptions(this->p_, &out));
   return SessionOptions{out};
 }
 
+template <typename T>
+inline std::string ConstSessionOptionsImpl<T>::GetConfigEntry(const char* config_key) const {
+  size_t size = 0;
+  // Feed nullptr for the data buffer to query the true size of the string value
+  Ort::ThrowOnError(GetApi().GetSessionConfigEntry(this->p_, config_key, nullptr, &size));
+
+  std::string out;
+  out.resize(size);
+  Ort::ThrowOnError(GetApi().GetSessionConfigEntry(this->p_, config_key, &out[0], &size));
```

Comment:
>&out[0] [](http://example.com/codeflow?start=73&length=7)

I think they added non-const `.data()` so wew no longer have to do this &[]

Reply:
I tried using `.data()` instead of `&out[0]`, but the [Linux CPU CI Pipeline](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=839463&view=logs&j=7536d2cd-87d4-54fe-4891-bfbbf2741d83&t=66420422-c7d6-5f71-625c-4b7851c9b9ba) fails because it is using an older gcc that does not support C++17.

```shell
/onnxruntime_src/js/node/../../include/onnxruntime/core/session/onnxruntime_cxx_inline.h: In function ‘void Ort::detail::attr_utils::GetAttr(const OrtKernelInfo*, const char*, std::string&)’:
/onnxruntime_src/js/node/../../include/onnxruntime/core/session/onnxruntime_cxx_inline.h:1529:77: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
 1529 |   Ort::ThrowOnError(GetApi().KernelInfoGetAttribute_string(p, name, out.data(), &size));
      |                                                                     ~~~~~~~~^~
      |                                                                             |
      |                                                                             const char*

```